src/time_advance.cpp

```diff
@@ -138,6 +139,19 @@ adaptive_advance(method const step_method, PDE<P> &pde,
       auto const y1 =
           adaptive_grid.redistribute_solution(y, old_plan, old_size);
       y.resize(y1.size()) = y1;
+
+      // after first refinement, save the refined vector to use as initial
+      // "guess" to GMRES
+      if (y_first_refine.empty())
+      {
+        y_first_refine = fk::vector<P>(y_refined);
```

Comment:
```suggestion
        y_first_refine = std::move(y_refined);
```
requires making `y_refined` mutable on line 120/121.